server/base/src/main/java/org/apache/accumulo/server/fs/VolumeChooserEnvironment.java

```diff
@@ -33,15 +34,29 @@
 
   private final ChooserScope scope;
   private final Table.ID tableId;
+  private final Text endRow;
 
   public VolumeChooserEnvironment(ChooserScope scope) {
     this.scope = Objects.requireNonNull(scope);
     this.tableId = null;
+    endRow = null;
   }
 
   public VolumeChooserEnvironment(Table.ID tableId) {
     this.scope = ChooserScope.TABLE;
     this.tableId = Objects.requireNonNull(tableId);
+    endRow = null;
+  }
+
+  public VolumeChooserEnvironment(Table.ID tableId, Text endRow) {
```

Comment:
Instead of just passing in the endRow, we could pass in a `TabletId` (`new TabletIdImpl(extent)`).
This would also cover the tableId, but for some reason that interface returns table IDs as a Text instead of the more convenient String (table IDs are always human readable). I wouldn't want to force people to convert to convert from Text to String to Table.ID, so I would probably leave that alone for now, and just add the extent as a `TabletId` public API type.

Reply:
Switched to the TabletId and pushed up.